src/socket_poller.cpp

```diff
@@ -149,12 +153,16 @@ int zmq::socket_poller_t::add_fd (fd_t fd_, void *user_data_, short events_)
     return 0;
 }
 
-int zmq::socket_poller_t::modify (socket_base_t  *socket_, short events_)
+int zmq::socket_poller_t::modify (socket_base_t  *socket_, void *zid,
+                                                           int zid_len,
+                                                           short events_)
 {
     items_t::iterator it;
 
     for (it = items.begin (); it != items.end (); ++it) {
-        if (it->socket == socket_)
+        if (it->socket != socket_)
+            continue;
+        if(it->zid_len == zid_len && memcmp(it->zid, zid, zid_len) == 0)
```

Comment:
missing whitespace in this line

Reply:
:+1: 

I did not check in detail style. Please note this is an RFC PR.